### PR TITLE
[go.mod] update libp2p module

### PR DESCRIPTION
**Pull request by LeoHChen**
Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

File: test/deploy.sh [hunk 1]
```diff
@@ -234,7 +234,7 @@ if [ "$DOTEST" == "true" ]; then
       i=$((i+1))
    done
    echo "waiting for the result"
-   sleep 20
```
Comment (rlan35) [inline test/deploy.sh]: why sleep more?

Comment (LeoHChen) [reply test/deploy.sh]: I simply found it is not enough to finish the cx, that makes the checking result failed.

**Comment by Daniel-VDM** [comment]
Jenkins, test this please

**Comment by LeoHChen** [comment]
@harmony-ek is also working on bring in new libp2p feature.  I'd just discard this PR.

**Pull request closed without merging**